charts/redpanda/templates/_helpers.tpl

```diff
@@ -305,6 +305,38 @@ Generate configuration needed for rpk
   {{- $result -}}
 {{- end -}}
 
+{{/*
+Returns the value of "resources.cpu.cores" in millicores.
+*/}}
+{{- define "redpanda-cores-in-millis" -}}
+  {{- $cores := .Values.resources.cpu.cores }}
+  {{- if $cores -}}
+    {{- if (hasSuffix "m" (toString $cores)) -}}
+      {{- trimSuffix "m" .Values.resources.cpu.cores | int -}}
+    {{- else -}}
+      {{- mulf 1000.0 ($cores | float64) | int -}}
+    {{- end -}}
+  {{- else -}}
+    {{ int "0" }}
+  {{- end -}}
+{{- end -}}
+
+{{/*
+Returns the SMP CPU count in whole cores, and sets "overprovisioned: true" when
+the "resources.cpu.cores" is less than 1 core.
+*/}}
+{{- define "redpanda-smp" -}}
+  {{- $coresInMillies := include "redpanda-cores-in-millis" . | int -}}
+  {{- if lt $coresInMillies 1000 }}
+    {{- $_ := set $.Values.resources.cpu "overprovisioned" true -}}
+  {{- end -}}
+  {{- if $coresInMillies -}}
+    {{- floor (divf $coresInMillies 1000) | int -}}
+  {{- else -}}
+    {{- int "1" -}}
+  {{- end -}}
+{{- end -}}
+
```

Comment:
nit: style `{{- ... -}}`

no need to `int "0"` as named templates always return strings you can just use a literal:
```
0
```

Does the change to $.Values persist outside of the named template? I've not tried that before, but I was pretty sure it didn't.

Reply:
> no need to int "0"

Sure thing, will update when I have a chance

> Does the change to `$.Values` persist outside of the named templat

Yes. It's magic ✨ (tested with `helm template`). The key is using the `$` context, afaik.

Reply:
> nit: style `{{- ... -}}`

Do you mean around returned value?

Reply:
For the nit: in named templates, unless there's a specific need for newlines, we always use the `-` both front and back.

Reply:
done, I think..
also, found some issues with defaulting logic